pkg/osquery/tables/execparsers/remotectl/parse.go

```diff
@@ -0,0 +1,128 @@
+//go:build darwin
+// +build darwin
+
+package remotectl
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+)
+
+// parseDumpstate parses the result of /usr/libexec/remotectl into a map that can be flattened by dataflatten.
+// We expect results in the following format, with empty newlines separating devices:
+//
+//	<device name>
+//			Key: Value
+//			Properties: {
+//				Key => Value
+//			}
+//			Services:
+//				service1
+//				service2
+func parseDumpstate(reader io.Reader) (any, error) {
```

Comment:
If I read this correctly, it's a not recursive -- it's assuming some things about the structure of the data. It looks like it's correctly parsing `Local Device`, but on my machine it goes a bit sideways parsing services under `Found ncm-0 (ncm-device)` (which is the iphone I have attached via usb)

The snippet of my output is:
```
	Services:
		com.apple.osanalytics.logTransfer
			Properties: {
				UsesRemoteXPC => true
			}
		com.apple.fusion.remote.service
			Version: 1
			Properties: {
				ServiceVersion => 1
				UsesRemoteXPC => true
			}
		com.apple.mobile.insecure_notification_proxy.shim.remote
		com.apple.sysdiagnose.remote
			Properties: {
				UsesRemoteXPC => true
			}
		com.apple.mobile.insecure_notification_proxy.remote
			Version: 1
			Properties: {
				ServiceVersion => 1
				UsesRemoteXPC => true
			}
		com.apple.mobile.lockdown.remote.untrusted
			Version: 1
			Properties: {
				ServiceVersion => 1
				UsesRemoteXPC => true
			}
	Local Services:
		com.apple.osanalytics.logRelay
```

And since that's all inside services it comes out  weirdly muddled with properties from that device.

Reply:
Though... If we didn't think some of those details mattered, we could stop processing. Maybe? I guess my inclination is usually to ingest everything we can

Reply:
I think grabbing the connected devices (eg. iPhones) is something I'd want the capability for if possible.

Reply:
Ooh, good to know, thanks -- I didn't have an example with output like this yet and couldn't find more docs on it. I'll update to handle this

Reply:
Full output with changes implemented --

```
osquery> select * from kolide_remotectl;
+------------------------------------------------------------+------------------------------------------+------------------------------+----------------------------------------------------------+-------+
| fullkey                                                    | parent                                   | key                          | value                                                    | query |
+------------------------------------------------------------+------------------------------------------+------------------------------+----------------------------------------------------------+-------+
| Local device/OS Build                                      | Local device                             | OS Build                     | 12.6 (21G115)                                            | *     |
| Local device/Product Type                                  | Local device                             | Product Type                 | MacBookPro18,1                                           | *     |
| Local device/Properties/AppleInternal                      | Local device/Properties                  | AppleInternal                | false                                                    | *     |
| Local device/Properties/BuildVersion                       | Local device/Properties                  | BuildVersion                 | 21G115                                                   | *     |
| Local device/Properties/CertificateSecurityMode            | Local device/Properties                  | CertificateSecurityMode      | true                                                     | *     |
| Local device/Properties/RestoreLongVersion                 | Local device/Properties                  | RestoreLongVersion           | 21.7.115.0.0,0                                           | *     |
| Local device/Properties/UniqueChipID                       | Local device/Properties                  | UniqueChipID                 | 6789012345678900                                         | *     |
| Local device/Properties/BridgeVersion                      | Local device/Properties                  | BridgeVersion                | 19.16.16067.0.0,0                                        | *     |
| Local device/Properties/DeviceEnclosureColor               | Local device/Properties                  | DeviceEnclosureColor         | 2                                                        | *     |
| Local device/Properties/EffectiveSecurityModeAp            | Local device/Properties                  | EffectiveSecurityModeAp      | true                                                     | *     |
| Local device/Properties/OSInstallEnvironment               | Local device/Properties                  | OSInstallEnvironment         | false                                                    | *     |
| Local device/Properties/HWModel                            | Local device/Properties                  | HWModel                      | J316sAP                                                  | *     |
| Local device/Properties/ProductName                        | Local device/Properties                  | ProductName                  | macOS                                                    | *     |
| Local device/Properties/RegionCode                         | Local device/Properties                  | RegionCode                   | LL                                                       | *     |
| Local device/Properties/SerialNumber                       | Local device/Properties                  | SerialNumber                 | 1234567890                                               | *     |
| Local device/Properties/UniqueDeviceID                     | Local device/Properties                  | UniqueDeviceID               | 11111111-012345789ABCDEF0                                | *     |
| Local device/Properties/OSVersion                          | Local device/Properties                  | OSVersion                    | 12.6                                                     | *     |
| Local device/Properties/BootSessionUUID                    | Local device/Properties                  | BootSessionUUID              | 00AE31FA-AAAA-BBBB-CCCC-DEADBEEFCAFE                     | *     |
| Local device/Properties/CPUArchitecture                    | Local device/Properties                  | CPUArchitecture              | arm64e                                                   | *     |
| Local device/Properties/DeviceClass                        | Local device/Properties                  | DeviceClass                  | Mac                                                      | *     |
| Local device/Properties/DeviceSupportsLockdown             | Local device/Properties                  | DeviceSupportsLockdown       | false                                                    | *     |
| Local device/Properties/EffectiveSecurityModeSEP           | Local device/Properties                  | EffectiveSecurityModeSEP     | true                                                     | *     |
| Local device/Properties/IsUIBuild                          | Local device/Properties                  | IsUIBuild                    | true                                                     | *     |
| Local device/Properties/MobileDeviceMinimumVersion         | Local device/Properties                  | MobileDeviceMinimumVersion   | 1351                                                     | *     |
| Local device/Properties/SensitivePropertiesVisible         | Local device/Properties                  | SensitivePropertiesVisible   | true                                                     | *     |
| Local device/Properties/CertificateProductionStatus        | Local device/Properties                  | CertificateProductionStatus  | true                                                     | *     |
| Local device/Properties/RegionInfo                         | Local device/Properties                  | RegionInfo                   | LL/A                                                     | *     |
| Local device/Properties/BoardId                            | Local device/Properties                  | BoardId                      | 10                                                       | *     |
| Local device/Properties/ChipID                             | Local device/Properties                  | ChipID                       | 24576                                                    | *     |
| Local device/Properties/HasSEP                             | Local device/Properties                  | HasSEP                       | true                                                     | *     |
| Local device/Properties/Image4Supported                    | Local device/Properties                  | Image4Supported              | true                                                     | *     |
| Local device/Properties/RemoteXPCVersionFlags              | Local device/Properties                  | RemoteXPCVersionFlags        | 72057594037927942                                        | *     |
| Local device/Properties/DeviceColor                        | Local device/Properties                  | DeviceColor                  | unknown                                                  | *     |
| Local device/Properties/EffectiveProductionStatusSEP       | Local device/Properties                  | EffectiveProductionStatusSEP | true                                                     | *     |
| Local device/Properties/HardwarePlatform                   | Local device/Properties                  | HardwarePlatform             | t6000                                                    | *     |
| Local device/Properties/EffectiveProductionStatusAp        | Local device/Properties                  | EffectiveProductionStatusAp  | true                                                     | *     |
| Local device/Properties/Image4CryptoHashMethod             | Local device/Properties                  | Image4CryptoHashMethod       | sha2-384                                                 | *     |
| Local device/Properties/ModelNumber                        | Local device/Properties                  | ModelNumber                  | Z14W00105                                                | *     |
| Local device/Properties/ProductType                        | Local device/Properties                  | ProductType                  | MacBookPro18,1                                           | *     |
| Local device/Properties/SecurityDomain                     | Local device/Properties                  | SecurityDomain               | 1                                                        | *     |
| Local device/Properties/SigningFuse                        | Local device/Properties                  | SigningFuse                  | true                                                     | *     |
| Local device/Services/0/Name                               | Local device/Services/0                  | Name                         | com.apple.remote.installcoordination_proxy               | *     |
| Local device/Services/1/Name                               | Local device/Services/1                  | Name                         | com.apple.sysdiagnose.remote                             | *     |
| Local device/Services/2/Name                               | Local device/Services/2                  | Name                         | com.apple.osanalytics.logRelay                           | *     |
| Local device/Services/3/Name                               | Local device/Services/3                  | Name                         | com.apple.corecaptured.remoteservice                     | *     |
| Local device/Services/4/Name                               | Local device/Services/4                  | Name                         | com.apple.mobile.storage_mounter_proxy.bridge            | *     |
| Local device/Services/5/Name                               | Local device/Services/5                  | Name                         | com.apple.dt.testmanagerd.remote.automation              | *     |
| Local device/Services/6/Name                               | Local device/Services/6                  | Name                         | com.apple.mobile.storage_mounter_proxy.bridge.macOS      | *     |
| Local device/Services/7/Name                               | Local device/Services/7                  | Name                         | com.apple.mobile.notification_proxy.remote               | *     |
| Local device/Services/8/Name                               | Local device/Services/8                  | Name                         | com.apple.dt.remoteFetchSymbols                          | *     |
| Local device/Services/9/Name                               | Local device/Services/9                  | Name                         | com.apple.CSCSupportd                                    | *     |
| Local device/Services/10/Name                              | Local device/Services/10                 | Name                         | ssh                                                      | *     |
| Local device/UUID                                          | Local device                             | UUID                         | 384B6869-AAAA-BBBB-CCCC-DEADBEEFCAFE                     | *     |
| Local device/Messaging Protocol Version                    | Local device                             | Messaging Protocol Version   | 3                                                        | *     |
| ncm-0 (ncm-device)/Heartbeat/0                             | ncm-0 (ncm-device)/Heartbeat             | 0                            | 0 heartbeats sent, 0 received                            | *     |
| ncm-0 (ncm-device)/Local Services/0/Name                   | ncm-0 (ncm-device)/Local Services/0      | Name                         | com.apple.osanalytics.logRelay                           | *     |
| ncm-0 (ncm-device)/Messaging Protocol Version              | ncm-0 (ncm-device)                       | Messaging Protocol Version   | 3                                                        | *     |
| ncm-0 (ncm-device)/OS Build                                | ncm-0 (ncm-device)                       | OS Build                     | 15.3.1 (19D52)                                           | *     |
| ncm-0 (ncm-device)/Properties/DeviceSupportsLockdown       | ncm-0 (ncm-device)/Properties            | DeviceSupportsLockdown       | true                                                     | *     |
| ncm-0 (ncm-device)/Properties/EffectiveProductionStatusSEP | ncm-0 (ncm-device)/Properties            | EffectiveProductionStatusSEP | true                                                     | *     |
| ncm-0 (ncm-device)/Properties/MobileDeviceMinimumVersion   | ncm-0 (ncm-device)/Properties            | MobileDeviceMinimumVersion   | 1351                                                     | *     |
| ncm-0 (ncm-device)/Properties/CertificateProductionStatus  | ncm-0 (ncm-device)/Properties            | CertificateProductionStatus  | true                                                     | *     |
| ncm-0 (ncm-device)/Properties/DeviceClass                  | ncm-0 (ncm-device)/Properties            | DeviceClass                  | iPhone                                                   | *     |
| ncm-0 (ncm-device)/Properties/DeviceEnclosureColor         | ncm-0 (ncm-device)/Properties            | DeviceEnclosureColor         | 1                                                        | *     |
| ncm-0 (ncm-device)/Properties/EffectiveProductionStatusAp  | ncm-0 (ncm-device)/Properties            | EffectiveProductionStatusAp  | true                                                     | *     |
| ncm-0 (ncm-device)/Properties/InterfaceIndex               | ncm-0 (ncm-device)/Properties            | InterfaceIndex               | 28                                                       | *     |
| ncm-0 (ncm-device)/Properties/ProductType                  | ncm-0 (ncm-device)/Properties            | ProductType                  | iPhone12,3                                               | *     |
| ncm-0 (ncm-device)/Properties/RegionCode                   | ncm-0 (ncm-device)/Properties            | RegionCode                   | LL                                                       | *     |
| ncm-0 (ncm-device)/Properties/BoardId                      | ncm-0 (ncm-device)/Properties            | BoardId                      | 6                                                        | *     |
| ncm-0 (ncm-device)/Properties/Image4CryptoHashMethod       | ncm-0 (ncm-device)/Properties            | Image4CryptoHashMethod       | sha2-384                                                 | *     |
| ncm-0 (ncm-device)/Properties/LocationID                   | ncm-0 (ncm-device)/Properties            | LocationID                   | 33333333                                                 | *     |
| ncm-0 (ncm-device)/Properties/RegionInfo                   | ncm-0 (ncm-device)/Properties            | RegionInfo                   | LL/A                                                     | *     |
| ncm-0 (ncm-device)/Properties/RemoteXPCVersionFlags        | ncm-0 (ncm-device)/Properties            | RemoteXPCVersionFlags        | 72057594037927942                                        | *     |
| ncm-0 (ncm-device)/Properties/SigningFuse                  | ncm-0 (ncm-device)/Properties            | SigningFuse                  | true                                                     | *     |
| ncm-0 (ncm-device)/Properties/StoreDemoMode                | ncm-0 (ncm-device)/Properties            | StoreDemoMode                | false                                                    | *     |
| ncm-0 (ncm-device)/Properties/BootSessionUUID              | ncm-0 (ncm-device)/Properties            | BootSessionUUID              | D4053160-AAAA-BBBB-CCCC-DEADBEEFCAFE                     | *     |
| ncm-0 (ncm-device)/Properties/CertificateSecurityMode      | ncm-0 (ncm-device)/Properties            | CertificateSecurityMode      | true                                                     | *     |
| ncm-0 (ncm-device)/Properties/SensitivePropertiesVisible   | ncm-0 (ncm-device)/Properties            | SensitivePropertiesVisible   | true                                                     | *     |
| ncm-0 (ncm-device)/Properties/SerialNumber                 | ncm-0 (ncm-device)/Properties            | SerialNumber                 | 1234567890XX                                             | *     |
| ncm-0 (ncm-device)/Properties/ChipID                       | ncm-0 (ncm-device)/Properties            | ChipID                       | 32816                                                    | *     |
| ncm-0 (ncm-device)/Properties/DeviceColor                  | ncm-0 (ncm-device)/Properties            | DeviceColor                  | 1                                                        | *     |
| ncm-0 (ncm-device)/Properties/EthernetMacAddress           | ncm-0 (ncm-device)/Properties            | EthernetMacAddress           | cc:d2:81:11:22:33                                        | *     |
| ncm-0 (ncm-device)/Properties/HardwarePlatform             | ncm-0 (ncm-device)/Properties            | HardwarePlatform             | t8030                                                    | *     |
| ncm-0 (ncm-device)/Properties/ModelNumber                  | ncm-0 (ncm-device)/Properties            | ModelNumber                  | MWA12                                                    | *     |
| ncm-0 (ncm-device)/Properties/ProductName                  | ncm-0 (ncm-device)/Properties            | ProductName                  | iPhone OS                                                | *     |
| ncm-0 (ncm-device)/Properties/UniqueDeviceID               | ncm-0 (ncm-device)/Properties            | UniqueDeviceID               | 11111111-0123456789ABCDF0                                | *     |
| ncm-0 (ncm-device)/Properties/BuildVersion                 | ncm-0 (ncm-device)/Properties            | BuildVersion                 | 19D52                                                    | *     |
| ncm-0 (ncm-device)/Properties/Image4Supported              | ncm-0 (ncm-device)/Properties            | Image4Supported              | true                                                     | *     |
| ncm-0 (ncm-device)/Properties/IsUIBuild                    | ncm-0 (ncm-device)/Properties            | IsUIBuild                    | true                                                     | *     |
| ncm-0 (ncm-device)/Properties/OSVersion                    | ncm-0 (ncm-device)/Properties            | OSVersion                    | 15.3.1                                                   | *     |
| ncm-0 (ncm-device)/Properties/SecurityDomain               | ncm-0 (ncm-device)/Properties            | SecurityDomain               | 1                                                        | *     |
| ncm-0 (ncm-device)/Properties/AppleInternal                | ncm-0 (ncm-device)/Properties            | AppleInternal                | false                                                    | *     |
| ncm-0 (ncm-device)/Properties/EffectiveSecurityModeSEP     | ncm-0 (ncm-device)/Properties            | EffectiveSecurityModeSEP     | true                                                     | *     |
| ncm-0 (ncm-device)/Properties/OSInstallEnvironment         | ncm-0 (ncm-device)/Properties            | OSInstallEnvironment         | false                                                    | *     |
| ncm-0 (ncm-device)/Properties/UniqueChipID                 | ncm-0 (ncm-device)/Properties            | UniqueChipID                 | 7890123456789ABC                                         | *     |
| ncm-0 (ncm-device)/Properties/CPUArchitecture              | ncm-0 (ncm-device)/Properties            | CPUArchitecture              | arm64e                                                   | *     |
| ncm-0 (ncm-device)/Properties/EffectiveSecurityModeAp      | ncm-0 (ncm-device)/Properties            | EffectiveSecurityModeAp      | true                                                     | *     |
| ncm-0 (ncm-device)/Properties/HWModel                      | ncm-0 (ncm-device)/Properties            | HWModel                      | D421AP                                                   | *     |
| ncm-0 (ncm-device)/Properties/HasSEP                       | ncm-0 (ncm-device)/Properties            | HasSEP                       | true                                                     | *     |
| ncm-0 (ncm-device)/Services/0/Name                         | ncm-0 (ncm-device)/Services/0            | Name                         | com.apple.mobile.insecure_notification_proxy.shim.remote | *     |
| ncm-0 (ncm-device)/Services/1/Properties/ServiceVersion    | ncm-0 (ncm-device)/Services/1/Properties | ServiceVersion               | 1                                                        | *     |
| ncm-0 (ncm-device)/Services/1/Properties/UsesRemoteXPC     | ncm-0 (ncm-device)/Services/1/Properties | UsesRemoteXPC                | true                                                     | *     |
| ncm-0 (ncm-device)/Services/1/Version                      | ncm-0 (ncm-device)/Services/1            | Version                      | 1                                                        | *     |
| ncm-0 (ncm-device)/Services/1/Name                         | ncm-0 (ncm-device)/Services/1            | Name                         | com.apple.mobile.insecure_notification_proxy.remote      | *     |
| ncm-0 (ncm-device)/Services/2/Name                         | ncm-0 (ncm-device)/Services/2            | Name                         | com.apple.osanalytics.logTransfer                        | *     |
| ncm-0 (ncm-device)/Services/2/Properties/UsesRemoteXPC     | ncm-0 (ncm-device)/Services/2/Properties | UsesRemoteXPC                | true                                                     | *     |
| ncm-0 (ncm-device)/Services/3/Name                         | ncm-0 (ncm-device)/Services/3            | Name                         | com.apple.fusion.remote.service                          | *     |
| ncm-0 (ncm-device)/Services/3/Properties/ServiceVersion    | ncm-0 (ncm-device)/Services/3/Properties | ServiceVersion               | 1                                                        | *     |
| ncm-0 (ncm-device)/Services/3/Properties/UsesRemoteXPC     | ncm-0 (ncm-device)/Services/3/Properties | UsesRemoteXPC                | true                                                     | *     |
| ncm-0 (ncm-device)/Services/3/Version                      | ncm-0 (ncm-device)/Services/3            | Version                      | 1                                                        | *     |
| ncm-0 (ncm-device)/Services/4/Name                         | ncm-0 (ncm-device)/Services/4            | Name                         | com.apple.mobile.lockdown.remote.untrusted               | *     |
| ncm-0 (ncm-device)/Services/4/Properties/UsesRemoteXPC     | ncm-0 (ncm-device)/Services/4/Properties | UsesRemoteXPC                | true                                                     | *     |
| ncm-0 (ncm-device)/Services/4/Properties/ServiceVersion    | ncm-0 (ncm-device)/Services/4/Properties | ServiceVersion               | 1                                                        | *     |
| ncm-0 (ncm-device)/Services/4/Version                      | ncm-0 (ncm-device)/Services/4            | Version                      | 1                                                        | *     |
| ncm-0 (ncm-device)/Services/5/Name                         | ncm-0 (ncm-device)/Services/5            | Name                         | com.apple.sysdiagnose.remote                             | *     |
| ncm-0 (ncm-device)/Services/5/Properties/UsesRemoteXPC     | ncm-0 (ncm-device)/Services/5/Properties | UsesRemoteXPC                | true                                                     | *     |
| ncm-0 (ncm-device)/Product Type                            | ncm-0 (ncm-device)                       | Product Type                 | iPhone12,3                                               | *     |
| ncm-0 (ncm-device)/State                                   | ncm-0 (ncm-device)                       | State                        | connected (connectable)                                  | *     |
| ncm-0 (ncm-device)/UUID                                    | ncm-0 (ncm-device)                       | UUID                         | 2C2D2409-AAAA-BBBB-CCCC-DEADBEEFCAFE                     | *     |
| ncm-1 (ncm-device)/Properties/BootSessionUUID              | ncm-1 (ncm-device)/Properties            | BootSessionUUID              | B8B27D90-AAAA-BBBB-CCCC-DEADBEEFCAFE                     | *     |
| ncm-1 (ncm-device)/Properties/HasSEP                       | ncm-1 (ncm-device)/Properties            | HasSEP                       | true                                                     | *     |
| ncm-1 (ncm-device)/Properties/OSInstallEnvironment         | ncm-1 (ncm-device)/Properties            | OSInstallEnvironment         | false                                                    | *     |
| ncm-1 (ncm-device)/Properties/OSVersion                    | ncm-1 (ncm-device)/Properties            | OSVersion                    | 16.0.2                                                   | *     |
| ncm-1 (ncm-device)/Properties/RegionInfo                   | ncm-1 (ncm-device)/Properties            | RegionInfo                   | LL/A                                                     | *     |
| ncm-1 (ncm-device)/Properties/SerialNumber                 | ncm-1 (ncm-device)/Properties            | SerialNumber                 | HRHX035W72                                               | *     |
| ncm-1 (ncm-device)/Properties/UniqueDeviceID               | ncm-1 (ncm-device)/Properties            | UniqueDeviceID               | 88888888-AABBCCDDEEFF0011                                | *     |
| ncm-1 (ncm-device)/Properties/ChipID                       | ncm-1 (ncm-device)/Properties            | ChipID                       | 33040                                                    | *     |
| ncm-1 (ncm-device)/Properties/ModelNumber                  | ncm-1 (ncm-device)/Properties            | ModelNumber                  | MLTW3                                                    | *     |
| ncm-1 (ncm-device)/Properties/RestoreLongVersion           | ncm-1 (ncm-device)/Properties            | RestoreLongVersion           | 20.1.380.0.0,0                                           | *     |
| ncm-1 (ncm-device)/Properties/StoreDemoMode                | ncm-1 (ncm-device)/Properties            | StoreDemoMode                | false                                                    | *     |
| ncm-1 (ncm-device)/Properties/DeviceEnclosureColor         | ncm-1 (ncm-device)/Properties            | DeviceEnclosureColor         | 1                                                        | *     |
| ncm-1 (ncm-device)/Properties/EffectiveProductionStatusSEP | ncm-1 (ncm-device)/Properties            | EffectiveProductionStatusSEP | true                                                     | *     |
| ncm-1 (ncm-device)/Properties/EffectiveSecurityModeAp      | ncm-1 (ncm-device)/Properties            | EffectiveSecurityModeAp      | true                                                     | *     |
| ncm-1 (ncm-device)/Properties/HWModel                      | ncm-1 (ncm-device)/Properties            | HWModel                      | D63AP                                                    | *     |
| ncm-1 (ncm-device)/Properties/Image4Supported              | ncm-1 (ncm-device)/Properties            | Image4Supported              | true                                                     | *     |
| ncm-1 (ncm-device)/Properties/ProductType                  | ncm-1 (ncm-device)/Properties            | ProductType                  | iPhone14,2                                               | *     |
| ncm-1 (ncm-device)/Properties/RegionCode                   | ncm-1 (ncm-device)/Properties            | RegionCode                   | LL                                                       | *     |
| ncm-1 (ncm-device)/Properties/SensitivePropertiesVisible   | ncm-1 (ncm-device)/Properties            | SensitivePropertiesVisible   | true                                                     | *     |
| ncm-1 (ncm-device)/Properties/CertificateProductionStatus  | ncm-1 (ncm-device)/Properties            | CertificateProductionStatus  | true                                                     | *     |
| ncm-1 (ncm-device)/Properties/DeviceColor                  | ncm-1 (ncm-device)/Properties            | DeviceColor                  | 1                                                        | *     |
| ncm-1 (ncm-device)/Properties/EffectiveProductionStatusAp  | ncm-1 (ncm-device)/Properties            | EffectiveProductionStatusAp  | true                                                     | *     |
| ncm-1 (ncm-device)/Properties/InterfaceIndex               | ncm-1 (ncm-device)/Properties            | InterfaceIndex               | 30                                                       | *     |
| ncm-1 (ncm-device)/Properties/ProductName                  | ncm-1 (ncm-device)/Properties            | ProductName                  | iPhone OS                                                | *     |
| ncm-1 (ncm-device)/Properties/AppleInternal                | ncm-1 (ncm-device)/Properties            | AppleInternal                | false                                                    | *     |
| ncm-1 (ncm-device)/Properties/DeviceClass                  | ncm-1 (ncm-device)/Properties            | DeviceClass                  | iPhone                                                   | *     |
| ncm-1 (ncm-device)/Properties/EthernetMacAddress           | ncm-1 (ncm-device)/Properties            | EthernetMacAddress           | dc:53:92:11:22:33                                        | *     |
| ncm-1 (ncm-device)/Properties/HardwarePlatform             | ncm-1 (ncm-device)/Properties            | HardwarePlatform             | t8110                                                    | *     |
| ncm-1 (ncm-device)/Properties/LocationID                   | ncm-1 (ncm-device)/Properties            | LocationID                   | 17825792                                                 | *     |
| ncm-1 (ncm-device)/Properties/SecurityDomain               | ncm-1 (ncm-device)/Properties            | SecurityDomain               | 1                                                        | *     |
| ncm-1 (ncm-device)/Properties/SigningFuse                  | ncm-1 (ncm-device)/Properties            | SigningFuse                  | true                                                     | *     |
| ncm-1 (ncm-device)/Properties/BoardId                      | ncm-1 (ncm-device)/Properties            | BoardId                      | 12                                                       | *     |
| ncm-1 (ncm-device)/Properties/BuildVersion                 | ncm-1 (ncm-device)/Properties            | BuildVersion                 | 20A380                                                   | *     |
| ncm-1 (ncm-device)/Properties/MobileDeviceMinimumVersion   | ncm-1 (ncm-device)/Properties            | MobileDeviceMinimumVersion   | 1400                                                     | *     |
| ncm-1 (ncm-device)/Properties/ThinningProductType          | ncm-1 (ncm-device)/Properties            | ThinningProductType          | iPhone14,2                                               | *     |
| ncm-1 (ncm-device)/Properties/UniqueChipID                 | ncm-1 (ncm-device)/Properties            | UniqueChipID                 | 67890ABCDEF1234                                          | *     |
| ncm-1 (ncm-device)/Properties/CPUArchitecture              | ncm-1 (ncm-device)/Properties            | CPUArchitecture              | arm64e                                                   | *     |
| ncm-1 (ncm-device)/Properties/CertificateSecurityMode      | ncm-1 (ncm-device)/Properties            | CertificateSecurityMode      | true                                                     | *     |
| ncm-1 (ncm-device)/Properties/DeviceSupportsLockdown       | ncm-1 (ncm-device)/Properties            | DeviceSupportsLockdown       | true                                                     | *     |
| ncm-1 (ncm-device)/Properties/RemoteXPCVersionFlags        | ncm-1 (ncm-device)/Properties            | RemoteXPCVersionFlags        | 72057594037927942                                        | *     |
| ncm-1 (ncm-device)/Properties/EffectiveSecurityModeSEP     | ncm-1 (ncm-device)/Properties            | EffectiveSecurityModeSEP     | true                                                     | *     |
| ncm-1 (ncm-device)/Properties/Image4CryptoHashMethod       | ncm-1 (ncm-device)/Properties            | Image4CryptoHashMethod       | sha2-384                                                 | *     |
| ncm-1 (ncm-device)/Properties/IsUIBuild                    | ncm-1 (ncm-device)/Properties            | IsUIBuild                    | true                                                     | *     |
| ncm-1 (ncm-device)/Services/0/Name                         | ncm-1 (ncm-device)/Services/0            | Name                         | com.apple.mobile.lockdown.remote.untrusted               | *     |
| ncm-1 (ncm-device)/Services/0/Properties/ServiceVersion    | ncm-1 (ncm-device)/Services/0/Properties | ServiceVersion               | 1                                                        | *     |
| ncm-1 (ncm-device)/Services/0/Properties/UsesRemoteXPC     | ncm-1 (ncm-device)/Services/0/Properties | UsesRemoteXPC                | true                                                     | *     |
| ncm-1 (ncm-device)/Services/0/Version                      | ncm-1 (ncm-device)/Services/0            | Version                      | 1                                                        | *     |
| ncm-1 (ncm-device)/Services/1/Name                         | ncm-1 (ncm-device)/Services/1            | Name                         | com.apple.mobile.insecure_notification_proxy.shim.remote | *     |
| ncm-1 (ncm-device)/Services/2/Name                         | ncm-1 (ncm-device)/Services/2            | Name                         | com.apple.mobile.insecure_notification_proxy.remote      | *     |
| ncm-1 (ncm-device)/Services/2/Properties/UsesRemoteXPC     | ncm-1 (ncm-device)/Services/2/Properties | UsesRemoteXPC                | true                                                     | *     |
| ncm-1 (ncm-device)/Services/2/Properties/ServiceVersion    | ncm-1 (ncm-device)/Services/2/Properties | ServiceVersion               | 1                                                        | *     |
| ncm-1 (ncm-device)/Services/2/Version                      | ncm-1 (ncm-device)/Services/2            | Version                      | 1                                                        | *     |
| ncm-1 (ncm-device)/Services/3/Name                         | ncm-1 (ncm-device)/Services/3            | Name                         | com.apple.osanalytics.logTransfer                        | *     |
| ncm-1 (ncm-device)/Services/3/Properties/UsesRemoteXPC     | ncm-1 (ncm-device)/Services/3/Properties | UsesRemoteXPC                | true                                                     | *     |
| ncm-1 (ncm-device)/Services/4/Properties/ServiceVersion    | ncm-1 (ncm-device)/Services/4/Properties | ServiceVersion               | 2                                                        | *     |
| ncm-1 (ncm-device)/Services/4/Properties/UsesRemoteXPC     | ncm-1 (ncm-device)/Services/4/Properties | UsesRemoteXPC                | true                                                     | *     |
| ncm-1 (ncm-device)/Services/4/Version                      | ncm-1 (ncm-device)/Services/4            | Version                      | 2                                                        | *     |
| ncm-1 (ncm-device)/Services/4/Name                         | ncm-1 (ncm-device)/Services/4            | Name                         | com.apple.internal.dt.coredevice.untrusted.tunnelservice | *     |
| ncm-1 (ncm-device)/Services/5/Version                      | ncm-1 (ncm-device)/Services/5            | Version                      | 1                                                        | *     |
| ncm-1 (ncm-device)/Services/5/Name                         | ncm-1 (ncm-device)/Services/5            | Name                         | com.apple.fusion.remote.service                          | *     |
| ncm-1 (ncm-device)/Services/5/Properties/UsesRemoteXPC     | ncm-1 (ncm-device)/Services/5/Properties | UsesRemoteXPC                | true                                                     | *     |
| ncm-1 (ncm-device)/Services/5/Properties/ServiceVersion    | ncm-1 (ncm-device)/Services/5/Properties | ServiceVersion               | 1                                                        | *     |
| ncm-1 (ncm-device)/UUID                                    | ncm-1 (ncm-device)                       | UUID                         | AAF6DCC0-AAAA-BBBB-CCCC-DEADBEEFCAFE                     | *     |
| ncm-1 (ncm-device)/Heartbeat/0                             | ncm-1 (ncm-device)/Heartbeat             | 0                            | 0 heartbeats sent, 0 received                            | *     |
| ncm-1 (ncm-device)/OS Build                                | ncm-1 (ncm-device)                       | OS Build                     | 16.0.2 (20A380)                                          | *     |
| ncm-1 (ncm-device)/Product Type                            | ncm-1 (ncm-device)                       | Product Type                 | iPhone14,2                                               | *     |
| ncm-1 (ncm-device)/State                                   | ncm-1 (ncm-device)                       | State                        | connected (connectable)                                  | *     |
| ncm-1 (ncm-device)/Local Services/0/Name                   | ncm-1 (ncm-device)/Local Services/0      | Name                         | com.apple.osanalytics.logRelay                           | *     |
| ncm-1 (ncm-device)/Messaging Protocol Version              | ncm-1 (ncm-device)                       | Messaging Protocol Version   | 3                                                        | *     |
+------------------------------------------------------------+------------------------------------------+------------------------------+----------------------------------------------------------+-------+
```